ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -165,12 +166,17 @@ initialChainSelection immutableDB volatileDB lgrDB tracer cfg varInvalid
     constructChains i succsOf = flip evalStateT Map.empty $
         mapM constructChain suffixesAfterI
       where
-        -- REVIEW We now prevent selecting more than k blocks in maximalCandidates
+        -- We now prevent selecting more than k blocks in maximalCandidates
         -- to avoid circumventing the LoE on startup.
-        -- Is this sensible?
-        -- Should we skip this if LoELimit is LoEUnlimited?
+        -- Shutting down a syncing node and then restarting it should not cause
+        -- it to select the longest chain the VolDB, since that chain might be
+        -- adversarial (ie the LoE did not allow the node to select it when it
+        -- arrived).
         suffixesAfterI :: [NonEmpty (HeaderHash blk)]
-        suffixesAfterI = Paths.maximalCandidates succsOf k (AF.anchorToPoint i)
+        suffixesAfterI = Paths.maximalCandidates succsOf limit (AF.anchorToPoint i)
+          where
+            limit | LoEDefault <- loELimit = k
+                  | otherwise = maxBound
```

Comment:
Very minor: do an exhaustive pattern match on `loELimit` to locally signal that `LoELimit` has only two constructors?